### PR TITLE
[BKY-5688] move executable testing out of the Makefile

### PR DESCRIPTION
**Pull request by mwittie**
## Describe your changes

I pushed the previous commit by mistake before merging all the local changes addressing PR feedback. This is the updated version

## Related Jira cards

https://blocky.atlassian.net/browse/BKY-5688

## Checklist before requesting a review

If any of these checks are missing, please provide an explanation.

- [x] I have updated README files, if applicable.
- [x] I have run `make pre-pr` and all checks have passed.
- [x] This PR is small. If not, justify below.

### CODE REVIEW

File: cmd/nitrite/main_test.go [hunk 1]
```diff
@@ -57,32 +59,64 @@ func runNitriteWithEnv(
...
-		assert.NotEmpty(t, outDoc)
-	})
+			// then
+			require.NoError(t, err)
+			assert.Equal(t, tt.time, outDoc.CreatedAt().UTC())
+			assert.Equal(t, tt.wantPCR0, base64.StdEncoding.EncodeToString(outDoc.PCRs[0]))
+			gotDebug, err := outDoc.Debug()
+			require.NoError(t, err)
+			assert.Equal(t, tt.debug, gotDebug)
+			assert.NotEmpty(t, outDoc.PublicKey)
```
Comment (tahardi) [inline cmd/nitrite/main_test.go]: The [nitro docs](https://docs.aws.amazon.com/enclaves/latest/user/verify-root.html#doc-def) state that the public key is an optional parameter. Given that it *could* be empty, should we actually be making this assertion?

**Review by tahardi — APPROVED** [review]
Looks good except for one question on asserting the public key

File: cmd/nitrite/main_test.go (hunk 1 above)
Comment (mwittie) [reply cmd/nitrite/main_test.go]: That's a good point. Dave suggested that we test the structure and content of the doc coming back and we do put a public key in all the attestations. Having said that, you're right, that its not required and so this test might not age well as a part of the library, where the testdata might change later on. I'll take this check out.

Comment (tahardi) [reply cmd/nitrite/main_test.go]: I thought we used "user data" for public keys (i.e., transitive attestation keys). I printed out the values and for both nitro attestations the public keys were an array of 1s while the self-signed attestation appeared to contain a legitimate public key.

**Pull request merged**